docs/book/src/component-config-tutorial/testdata/project/config/rbac/role.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```

Comment:
@camilamacedo86 I'm confused on the need of cluster role separately before the crd and roles are created by controller-gen. Do we have an issue for this, to get more context? Currently, KB doesn't support fetching core types and not generating CR. If so, why do we need to create manager's RBAC preemptively?

Reply:
Hi @varshaprasad96,

Thanks for your review and your help on this one

- controller-gen does indeed target config/rbac/role.yaml and not clusterroles (that must be scaffiold by kubebuilder), but I do not think that has a relation with the changes of this PR.
I will try better clarify: 

- The crux is: that not all projects will require CRDs. So, enabling CRDs by default in the scaffold might not be appropriate. Note that a project can exist without the need for its own APIs at all.
- This PR modifies the initial state of config/default/kustomization.yaml to comment out CRDs. [See the change](https://github.com/kubernetes-sigs/kubebuilder/pull/3627/files#diff-20b19ff8cca874cadba14de74e485c68d13fd42df3eb0e9dcb42decd4b442619).
If an API is created later, the CRD section will be uncommented automatically.

Hope this makes the PR's intention clear. Let me know if you have further questions!

Reply:
Thanks for the detailed reply, @camilamacedo86! This makes sense and answers my question.